### PR TITLE
Template for barebones go provider

### PR DESCRIPTION
**Pull request by jordan-rash**
## Feature or Problem
Template for barebones provider using golang tooling

### CODE REVIEW

File: provider/barebones-go/.goreleaser.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    goarch:
+      - arm64
+      - amd64
+    goarm:
+      - "7"
+    ignore:
+      - goos: windows
+        goarch: arm
+      - goos: windows
+        goarch: arm64
```
Comment (brooksmtownsend) [inline provider/barebones-go/.goreleaser.yml:20]: sad windows noises

File: provider/barebones-go/project-generate.toml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+[[placeholders]]
+name = "capability_id"
+type = "string"
+prompt = "Capability ID?"
+default = "wasmcloud:httpserver"
+
+[[placeholders]]
+name = "go_mod_prefix"
+type = "string"
+prompt = "Go Module name prefix (will add name to end)?"
```
Comment (brooksmtownsend) [inline provider/barebones-go/project-generate.toml:21]: nit: maybe the ? would be better before the parenthesis? I read it wrong but not sure if that's just me

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]
Only real requested change is to add a README with some information about this provider. It would also be great to outline the steps to add a handler to implement a contract, since for me I don't know what I'd have to add. 

Thanks for adding this!

**Comment by jordan-rash** [comment]
> It would also be great to outline the steps to add a handler to implement a contract, since for me I don't know what I'd have to add.

Would it be better to add a real provider example?  Maybe the httpserver?  This is template is just meant for a quick start....

> Only real requested change is to add a README with some information about this provider
This really isnt a "provider" in its current state, but to your point, i will def get a readme saying that

File: provider/barebones-go/.goreleaser.yml (hunk 1 above)
Comment (jordan-rash) [reply provider/barebones-go/.goreleaser.yml:20]: This should still be building windows x86....i just realized I didnt add it to the par release tho, ill add that

**Review by brooksmtownsend — APPROVED** [review]

**Pull request merged**